contracts/IRelayHub.sol

```diff
@@ -113,17 +113,21 @@ contract IRelayHub {
     // Emits a TransactionRelayed event.
     function relayCall(address from, address to, bytes memory encodedFunction, uint256 transactionFee, uint256 gasPrice, uint256 gasLimit, uint256 nonce, bytes memory approval) public;
 
+    // Emitted when an attempt to relay a call failed. This can happen due to incorrect relayCall arguments, or the
+    // recipient not accepting the relayed call. The actual relayed call was not executed, and the recipient not charged.
+    // The reason field contains an error code: values 1-10 correspond to PreconditionCheck entries, and values over 10
+    // are custom recipient error codes returned from acceptRelayedCall.
+    event RelayCallFailed(address indexed relay, address indexed from, address indexed to, bytes4 selector, uint256 reason);
```

Comment:
I think we can do with this events model, just fix the event name: its merely means that `CanRelayFailed`, and should be named so...